camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -216,16 +217,18 @@ public void run() throws Exception
       try
       {
         runCollectorForTopicDir(fs, topicName, new Path(topic.getPath(), sourceSubdir), destinationPath);
+        log.info("Shutting down priority executor");
+        executorService.shutdown();
```

Comment:
Yes otherwise we get into a dead lock. Because JVM cannot shutdown with unfinished executors.